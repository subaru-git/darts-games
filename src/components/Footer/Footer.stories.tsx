import type { Meta, StoryObj } from '@storybook/react';
import Footer from './Footer';

const meta: Meta<typeof Footer> = { component: Footer, title: 'Components/Footer' };

export default meta;
type Story = StoryObj<typeof Footer>;
export const Default: Story = {};
