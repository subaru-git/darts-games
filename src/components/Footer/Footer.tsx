import React, { FC } from 'react';
import Link from 'next/link';
import { AiFillGithub, AiOutlineTwitter } from 'react-icons/ai';
import useLocale from '@/hooks/locale';

const Footer: FC = () => {
  const { t } = useLocale();
  return (
    <div className='absolute bottom-0 -mb-80 h-80 w-full bg-gray-700 md:-mb-52 md:h-52'>
      <div className='mt-2 flex flex-col items-center md:mt-4 md:gap-4'>
        <div className='flex flex-1 flex-col justify-center gap-6 p-4 md:flex-row md:gap-8'>
          <div className='flex flex-col gap-2 md:basis-48'>
            <h2 className='text-xl font-bold text-white'>{`Darts Homes`}</h2>
            <p className='text-xs text-white'>{t.footer.description}</p>
          </div>
          <div className='flex flex-col gap-2 md:basis-48'>
            <h2 className='text-xl font-bold text-white'>{`Feedback`}</h2>
            <div className='flex gap-3'>
              <a
                href={'https://github.com/subaru-git/darts-homes'}
                target='_blank'
                rel='noopener noreferrer'
                aria-label={'subaru-git github page'}
              >
                <AiFillGithub size={24} color={'white'} />
              </a>
              <a
                href={'https://twitter.com/darts_homes'}
                target='_blank'
                rel='noopener noreferrer'
                aria-label={'darts_homes twitter'}
              >
                <AiOutlineTwitter size={24} color={'white'} />
              </a>
            </div>
          </div>
          <div className='flex flex-col gap-2 md:basis-48'>
            <h2 className='text-xl font-bold text-white'>{`Legal`}</h2>
            <div className='text-xs text-white'>
              <Link href={'/termsofservice'}>{t.footer.terms}</Link>
            </div>
            <div className='text-xs text-white'>
              <Link href={'/privacypolicy'}>{t.footer.privacy}</Link>
            </div>
          </div>
        </div>
        <div className='text-xs text-gray-400 md:p-5'>{`Copyright Â© 2024 Darts Homes`}</div>
      </div>
    </div>
  );
};

export default Footer;
