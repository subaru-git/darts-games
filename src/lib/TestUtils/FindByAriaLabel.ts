export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
