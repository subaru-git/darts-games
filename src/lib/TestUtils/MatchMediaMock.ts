import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
