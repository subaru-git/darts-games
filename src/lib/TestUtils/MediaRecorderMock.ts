export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
