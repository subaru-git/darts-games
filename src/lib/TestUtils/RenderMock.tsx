import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';
import { NextIntlClientProvider } from 'next-intl';

export const render = (children: ReactNode | ReactNode[]) => {
  const messages = require('@/locales/en.json');
  return defaultRender(
    <ChakraProvider>
      <NextIntlClientProvider messages={messages} locale='en'>
        <Providers>{children}</Providers>
      </NextIntlClientProvider>
    </ChakraProvider>,
  );
};
