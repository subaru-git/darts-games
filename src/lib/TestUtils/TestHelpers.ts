import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}