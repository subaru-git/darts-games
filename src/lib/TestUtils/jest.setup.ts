import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
