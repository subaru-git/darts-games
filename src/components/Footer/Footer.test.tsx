import { screen } from '@testing-library/react';
import Footer from './Footer';
import { render } from '@/lib/TestUtils/RenderMock';

test('should rendering', () => {
  const { container } = render(<Footer />);
  expect(screen.getAllByText('Copyright Â© 2024 Darts Homes')).toHaveLength(1);
  expect(container).toMatchSnapshot();
});
