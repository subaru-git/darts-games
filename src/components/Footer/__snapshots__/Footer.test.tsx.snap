// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="absolute bottom-0 -mb-80 h-80 w-full bg-gray-700 md:-mb-52 md:h-52"
  >
    <div
      class="mt-2 flex flex-col items-center md:mt-4 md:gap-4"
    >
      <div
        class="flex flex-1 flex-col justify-center gap-6 p-4 md:flex-row md:gap-8"
      >
        <div
          class="flex flex-col gap-2 md:basis-48"
        >
          <h2
            class="text-xl font-bold text-white"
          >
            Darts Homes
          </h2>
          <p
            class="text-xs text-white"
          >
            It supports home darts, which tend to be monotonous.
          </p>
        </div>
        <div
          class="flex flex-col gap-2 md:basis-48"
        >
          <h2
            class="text-xl font-bold text-white"
          >
            Feedback
          </h2>
          <div
            class="flex gap-3"
          >
            <a
              aria-label="subaru-git github page"
              href="https://github.com/subaru-git/darts-homes"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                color="white"
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 1024 1024"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                />
              </svg>
            </a>
            <a
              aria-label="darts_homes x"
              href="https://twitter.com/darts_homes"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                color="white"
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 512 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="flex flex-col gap-2 md:basis-48"
        >
          <h2
            class="text-xl font-bold text-white"
          >
            Legal
          </h2>
          <div
            class="text-xs text-white"
          >
            <a
              href="/termsofservice"
            >
              Terms of service (Japanese Only)
            </a>
          </div>
          <div
            class="text-xs text-white"
          >
            <a
              href="/privacypolicy"
            >
              Privacy Policy (Japanese Only)
            </a>
          </div>
        </div>
      </div>
      <div
        class="text-xs text-gray-400 md:p-5"
      >
        Copyright Â© 2024 Darts Homes
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
